mam4_refactor_scripts/test_gen_for_sub.sh

```diff
@@ -0,0 +1,114 @@
+#!/bin/sh
+
+
+main() {
+    #-------------------------
+    #User input Starts
+    #-------------------------
+
+    #subroutine name
+    sub_name=mam_rename_1subarea
+
+    #module name
+    module_name=modal_aero_rename
+
+    #file path
+    dir_path=components/eam/src/chemistry/modal_aero/
+
+    #CPP directive to turn on file writing
+    cpp_directive=YAML_RENAME
+
+    #-------------------------
+    #USER INPUT ENDS
+    #-------------------------
+
+    #----------------------------------------------------------------------------------------------------
+    #----------------------------------------------------------------------------------------------------
+
+
+    #ensure that script is run in the right directory (i.e. "e3sm_mam4_refactor") as
+    #relative path code works only if script is run at the code root
+    this_dir=${PWD##*/}
+
+    if [ $this_dir != "e3sm_mam4_refactor" ]; then
+        echo 'Please run this script at source code root in directory "e3sm_mam4_refactor"'
+        exit 1
+    fi
+
+    #Yaml directory path
+    yaml_dir_path=components/eam/src/chemistry/yaml
+
+    #path to directory containing beg and end input stubs
+    stub_dir=mam4_refactor_scripts
+
+
+
+    #relative path
+    relative_path=`realpath --relative-to="$dir_path" "$yaml_dir_path"`
+
+    #create directory and files if not already created
+    /bin/mkdir -p $yaml_dir_path/$module_name/f90_yaml
+
+    #code to add
+    echo '-------------'
+    echo 'Code to add:'
+    echo '-------------'
+
+    echo '(If not already added, add the following line at module level at the top)'
+    echo "#include \"$relative_path/common_files/common_uses.ymlf90\""
+
+    #Subroutine beginning file
+    newline
+    create_file $module_name $sub_name $yaml_dir_path $relative_path "beginning"
+
+    #subroutine end file
+    newline
+    create_file $module_name $sub_name $yaml_dir_path $relative_path "end"
+}
+
+#---------------------
+# Function Definitions
+#---------------------
+
+#Generate a newline
+newline () { echo ''; }
+
+create_file () {
+
+    #create_file $module_name $sub_name $yaml_dir_path $relative_path "end"
+    echo "(Add the following line at the $5 of the subroutine)"
+
+    #find beginning or end sub string
+    if [ $5 == "beginning" ]; then
+        sub_str=beg
+    elif [ $5 == "end" ]; then
+        sub_str=end
+    else
+        echo 'Invalid string (beginning or end)for function create_file'
+        exit 1
+    fi
+
+    #full file name
+    f_name=${sub_name}_${sub_str}.ymlf90
```

Comment:
I use emacs and I added this extension in my `.emacs` file so that it is recognized as an F90 file. I am not sure if something like that exists for the vi editor. We can actually make the extensions `.f90`. I was initially worried that E3SM might compile these files even in the default mode (which is extra time for compilation). I recently tested and found that E3SM doesn't actually do that. I will change the extensions to `f90` for future files.